### ISSUE TITLE
App gets crashed while updating to 2.8.3, it works fine on 2.8.2

### ISSUE THREAD

**Issue by ZubairAkber** [aws-amplify/amplify-android#2449]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

DataStore

### Gradle script dependencies

<details>

    // Amplify dependencies
    implementation 'com.amplifyframework:core:2.8.3'
    implementation 'com.amplifyframework:aws-api:2.8.3'
    implementation 'com.amplifyframework:aws-datastore:2.8.3'
    implementation 'com.amplifyframework:aws-auth-cognito:2.8.3'
    implementation 'com.amplifyframework:aws-storage-s3:2.8.3'

    // AWS MQTT dependencies
    implementation 'com.amazonaws:aws-android-sdk-iot:2.69.0'
    implementation 'com.amazonaws:aws-android-sdk-mobile-client:2.69.0'

</details>

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

When i move to version 2.8.3 app gets crashed, it worked fine on 2.8.2 version of Amplify

### Reproduction steps (if applicable)

Simply update to 2.8.3 version

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line

aws-datastore    W  Reading subscriptions buffer has failed.
                                         java.lang.IllegalStateException: Only a single observer allowed.
                                         	at io.reactivex.rxjava3.subjects.UnicastSubject.subscribeActual(UnicastSubject.java:294)
                                         	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
                                         	at io.reactivex.rxjava3.internal.operators.observable.ObservableDoOnLifecycle.subscribeActual(ObservableDoOnLifecycle.java:33)
                                         	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
                                         	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable.subscribeActual(ObservableFlatMapCompletableCompletable.java:49)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2836)
                                         	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.startDrainingMutationBuffer(SubscriptionProcessor.java:258)
                                         	at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$4$com-amplifyframework-datastore-syncengine-Orchestrator(Orchestrator.java:351)
                                         	at com.amplifyframework.datastore.syncengine.Orchestrator$$ExternalSyntheticLambda8.subscribe(Unknown Source:2)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
                                         	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
                                         	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
                                         	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
                                         	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                                         	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
                                         	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                         	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                         	at java.lang.Thread.run(Thread.java:923)
                                         	
                                         	
AndroidRuntime           E  FATAL EXCEPTION: RxCachedThreadScheduler-40
                                         Process: com.mtronic.smarthome, PID: 27889
                                         io.reactivex.rxjava3.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalStateException: Only a single observer allowed.
                                         	at io.reactivex.rxjava3.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:95)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:95)
                                         	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:156)
                                         	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable$FlatMapCompletableMainObserver.onError(ObservableFlatMapCompletableCompletable.java:125)
                                         	at io.reactivex.rxjava3.internal.observers.DisposableLambdaObserver.onError(DisposableLambdaObserver.java:65)
                                         	at io.reactivex.rxjava3.internal.disposables.EmptyDisposable.error(EmptyDisposable.java:63)
                                         	at io.reactivex.rxjava3.subjects.UnicastSubject.subscribeActual(UnicastSubject.java:294)
                                         	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
                                         	at io.reactivex.rxjava3.internal.operators.observable.ObservableDoOnLifecycle.subscribeActual(ObservableDoOnLifecycle.java:33)
                                         	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
                                         	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable.subscribeActual(ObservableFlatMapCompletableCompletable.java:49)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2836)
                                         	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.startDrainingMutationBuffer(SubscriptionProcessor.java:258)
                                         	at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$4$com-amplifyframework-datastore-syncengine-Orchestrator(Orchestrator.java:351)
                                         	at com.amplifyframework.datastore.syncengine.Orchestrator$$ExternalSyntheticLambda8.subscribe(Unknown Source:2)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
                                         	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
                                         	at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
                                         	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
                                         	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
                                         	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
                                         	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
15:54:01.109 AndroidRuntime           E  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
                                         	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                         	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                         	at java.lang.Thread.run(Thread.java:923)
                                         Caused by: java.lang.IllegalStateException: Only a single observer allowed.
                                         	... 31 more

</details>

**Comment by eeatonaws** [comment]
A fix for the issue has been released in Amplify Android 2.8.4.

**Comment by antoineberthelin** [comment]
@eeatonaws @ZubairAkber Release 2.8.4 is very good for me, my load test are OK 👍 
Thanks
